scripts/dev/pre-commit.validate-source

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+###############################################################################
+# File  : script/developer/pre-commit.validate-source
+#
+# Called by $SCALE/.git/hooks/pre-commit to apply various preliminary checks
+###############################################################################
```

Comment:
```suggestion
#!/bin/sh -e
# Copyright 2021-2024 UT-Battelle, LLC, and other Celeritas developers.
# See the top-level COPYRIGHT file for details.
# SPDX-License-Identifier: (Apache-2.0 OR MIT)
```

Reply:
I didn't set `-e` because `rg`/`grep` return `1` if they don't find any match and we don't want to fail if that's the case. Does `sh` always supports read into an array variable?

Reply:
Not sure, but `bash` is very old on modern mac systems and may be eventually deleted...

Reply:
Actually, `-e` has no effect currently because of how the function is called before a `||`: `check_corecel_dependencies || exit_with_message "..."`

Reply:
Yeah I usually run with `-e` because if some other part of teh script has an unexpected error, the program continues in a rather undefined state...